### PR TITLE
KAFKA-10070: parameterize Connect unit tests to remove code duplication

### PR DESCRIPTION
**Pull request by levzem**
Signed-off-by: Lev Zemlyanov <lev@confluent.io>

Parameterize Connect unit tests to remove code duplication. Some PRs added duplicate test classes to test topic creation. This fix parameterizes the topic creation to reduce code duplication.

The topic creation tests were copied over to the original test classes to capture any changes to reduce risk of test coverage.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by levzem** [comment]
@C0urante @kkonstantine @rhauch would appreciate some 👀 , thanks!

File: connect/runtime/src/test/java/org/apache/kafka/connect/util/ParameterizedTest.java
Comment (mimaison) [inline connect/runtime/src/test/java/org/apache/kafka/connect/util/ParameterizedTest.java:17]: We're missing the license header here

File: checkstyle/suppressions.xml
Comment (mimaison) [inline checkstyle/suppressions.xml]: Instead of adding this to Kafka Streams rules, can we move it to the Connect section just above?

File: connect/runtime/src/test/java/org/apache/kafka/connect/runtime/ErrorHandlingTaskTest.java
Comment (mimaison) [inline connect/runtime/src/test/java/org/apache/kafka/connect/runtime/ErrorHandlingTaskTest.java]: Should we move this with the other `java.util` imports below?

File: connect/runtime/src/test/java/org/apache/kafka/connect/util/ParameterizedTest.java
Comment (mimaison) [inline connect/runtime/src/test/java/org/apache/kafka/connect/util/ParameterizedTest.java]: Checkstyle fails because it expects indentations to be 4 spaces

**Review by mimaison — CHANGES_REQUESTED** [review]
Thanks for the PR. That's a pretty nice cleanup!
I've only taken a quick look so far but there are a few small formatting issues that are failing checkstyle

**Comment by levzem** [comment]
tests failing are unrelated to the changes

`Build / JDK 11 / kafka.api.PlaintextConsumerTest.testAutoCommitOnCloseAfterWakeup()`
`Build / JDK 15 / org.apache.kafka.connect.integration.RebalanceSourceConnectorsIntegrationTest.testMultipleWorkersRejoining`

@mimaison thanks for the review, mind taking another pass?

**Review by kkonstantine — APPROVED** [review]
Thanks @levzem 
Glad to see there's a workaround to make parameterized tests work with powermock. 

LGTM, but I'll wait for @mimaison's final approval.

**Review by mimaison — APPROVED** [review]

**Pull request merged**
